Added Matter bridge application (#11880)
nrfconnect/sdk-nrf: applications/matter_bridge/README.rst

.. parsed-literal::
   :class: highlight

   west build -b nrf7002dk_nrf5340_cpuapp -d build_nrf7002dk_nrf5340_cpuapp -- -DCONF_FILE=prj_release.conf

The ``build_nrf7002dk_nrf5340_cpuapp`` parameter specifies the output directory for the build files.

.. note::
   If the selected board does not support the selected build type, the build is interrupted.
   For example, if the ``shell`` build type is not supported by the selected board, the following notification appears:

   .. code-block:: console

      File not found: ./ncs/nrf/applications/matter_bridge/configuration/nrf7002dk_nrf5340_cpuapp/prj_shell.conf

Testing
=======

After building the sample and programming it to your development kit, complete the following steps to test its basic features:

#. |connect_kit|
#. |connect_terminal_ANSI|
#. Observe that the **LED1** starts flashing (short flash on).
   This means that the sample has automatically started the Bluetooth LE advertising.
#. Commission the device to the Matter network.
   See `Commissioning the device`_ for more information.

>    During the commissioning process, write down the value for the bridge node ID, the temperature sensor node ID that is going to be used in the next steps (*<bridge_node_ID>*)

Author: It's a copy paste error. There should be only bridge node ID.

  You can use this version to enable additional features for verifying the application behavior, such as logs.
* ``release`` -- Release version of the application.
  You can use this version to enable only the necessary application functionalities to optimize its performance.

.. note::
    `Selecting a build type`_ is optional.
    The ``debug`` build type is used by default if no build type is explicitly selected.

User interface
**************

.. include:: ../../samples/matter/lock/README.rst
    :start-after: matter_door_lock_sample_led1_start
    :end-before: matter_door_lock_sample_led1_end

Button 1:
     If pressed for six seconds, it initiates the factory reset of the device.
     Releasing the button within the six-second window cancels the factory reset procedure.

Button 2:
     Enables Bluetooth LE advertising for the predefined period of time (15 minutes by default), and makes the device discoverable over Bluetooth LE.

.. include:: ../lock/README.rst
    :start-after: matter_door_lock_sample_jlink_start
    :end-before: matter_door_lock_sample_jlink_end

.. _matter_bridge_cli:

Matter CLI commands
===================

> You can use the following commands to control the bridge device:
> 

Reviewer: In the doc output, it will be hard to distinguish information about the two commands. I suggest you add lines 108-130 in a bullet point, and lines 132-146 in another. You can have:
```suggestion
You can use the following commands to control the bridge device:

* For adding a new bridged device to the bridge:

...

* For removes a new bridged device from the bridge:

```

Author: Ok, I think it was like this before the review from @wiba-nordic, so can we have some agreement on that?

Author: Or maybe not, it wasn't the same. Ok I will introduce it :)

Author: Ok, done, I hope it is correct now.

> ::

Reviewer: ```suggestion
.. code-block:: console
```

   matter_bridge add <device_type> [node_label]

This command adds a new bridged device to the bridge.
In this command:

* ``<device_type>`` is a Matter device type to use for the bridged device.
  The argument is mandatory and accepts the following values:

   * ``256 - On/Off Light.
   * ``770`` - Temperature Sensor.
   * ``775`` - Humidity Sensor.

* ``[node label]`` is a node label for the bridged device.
  The argument is optional and you can use it to better personalize the device in your application.
   
For example:

.. parsed-literal::
   :class: highlight

>    uart:~$ matter_bridge add 256 "Kitchen Light"

Reviewer 2: Add a line that displays the assigned endpoint ID so that it's clear how a user knows the endpoint ID for the "remove" command?

> ::

Reviewer: ```suggestion
.. code-block:: console
```

   matter_bridge remove <bridged_device_endpoint_id>

> This command removes a new bridged device from the bridge.

Reviewer: I would drop "new" here, because this would suggest you can only remove the last added bridge.

In this command:

>    * ``<bridged_device_endpoint_id>`` is the endpoint id of the bridged device to be removed

Reviewer: ```suggestion
   * ``<bridged_device_endpoint_id>`` is the endpoint ID of the bridged device to be removed.
```

.. _matter_bridge_app:

> Matter: Bridge
> ################

Reviewer: ```suggestion
Matter bridge
##############
```
I would treat this on-par with "Matter controller" and other network topology devices.
See https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/matter/overview/network_topologies.html#matter-network-concepts.
Also because, assuming that "Matter:" is here to indicate the protocol and the second part is the device type, this could be interpreted we're adding an application for controlling bridges. ;)

    The ``debug`` build type is used by default if no build type is explicitly selected.

User interface
**************

.. include:: ../../samples/matter/lock/README.rst
    :start-after: matter_door_lock_sample_led1_start
    :end-before: matter_door_lock_sample_led1_end

Button 1:
     If pressed for six seconds, it initiates the factory reset of the device.
     Releasing the button within the six-second window cancels the factory reset procedure.

Button 2:
     Enables Bluetooth LE advertising for the predefined period of time (15 minutes by default), and makes the device discoverable over Bluetooth LE.

.. include:: ../../samples/matter/lock/README.rst
    :start-after: matter_door_lock_sample_jlink_start
    :end-before: matter_door_lock_sample_jlink_end

.. _matter_bridge_cli:

Matter CLI commands
===================

You can use the following commands to control the bridge device:

* For adding a new bridged device to the bridge:

   .. code-block:: console

      matter_bridge add <device_type> [node_label]

   This command adds a new bridged device to the bridge.
   In this command:

   * ``<device_type>`` is a Matter device type to use for the bridged device.

>    The argument is mandatory and accepts the following values:

Reviewer 3: ```suggestion
     The argument is mandatory and accepts the following values:
```
Wrong indent breaks doc build.

      * ``256`` - On/Off Light.
      * ``770`` - Temperature Sensor.
      * ``775`` - Humidity Sensor.

   * ``[node label]`` is a node label for the bridged device.
     The argument is optional and you can use it to better personalize the device in your application.

   For example:

   .. parsed-literal::
      :class: highlight

      uart:~$ matter_bridge add 256 "Kitchen Light"

   You should be able to see the following output on the terminal:

      .. parsed-literal::
         :class: highlight

         I: Adding OnOff Light bridged device
         I: Added device to dynamic endpoint 3 (index=0)

> * For removes a new bridged device from the bridge:

Reviewer 3: ```suggestion
* For removing a bridged device from the bridge:
```

Onboarding information
----------------------

When you start the commissioning procedure, the controller must get the onboarding information from the Matter accessory device.
The onboarding information representation depends on your commissioner setup.

> For this sample, you can use one of the following :ref:`onboarding information formats <ug_matter_network_topologies_commissioning_onboarding_formats>` to provide the commissioner with the data payload that includes the device discriminator and the setup PIN code:

Reviewer 3: ```suggestion
For this application, you can use one of the following :ref:`onboarding information formats <ug_matter_network_topologies_commissioning_onboarding_formats>` to provide the commissioner with the data payload that includes the device discriminator and the setup PIN code:
```
